### PR TITLE
Updates SpiceRotation to use ale::Orientations

### PR DESCRIPTION
**Pull request by krlberry**
## Description
Updates SpiceRotation to use ale::Orientations. 

Notably only uses Orientations to to calculate/store `p_CJ` (time-dependent rotation) and `p_av` (angular velocity) from SpiceRotation. Since the constant and non-constant frames and constant rotation are also part of the ale::Orientations object, it would have been possible to store those in the m_orientations object, too, but I didn't want to be creating new Orientations objects constantly to update/add a list of time-dependent frames, etc... 

Some tests will fail because this requires a new version of ale to run. 

## Related Issue
Ale work, no ticket.

## Motivation and Context
Use one library (ale) to do calculations for both ISIS and USGSCSM. 

## How Has This Been Tested?
SpiceRotation unit test matches, except for some very small numbers that are basically 0 vs. actual 0 in the output.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/SpiceRotation/SpiceRotation.cpp [hunk 1]
```diff
@@ -373,12 +384,29 @@ namespace Isis {
     int cacheSize = p_cacheTime.size();
 
     // Loop and load the cache
+    std::vector<ale::Rotation> rotationCache;
+    std::vector< std::vector<double> > cache;
```
Comment (jessemapel) [inline isis/src/base/objs/SpiceRotation/SpiceRotation.cpp:388]: Why is this cache here? it doesn't look like it gets used.

File: isis/src/base/objs/SpiceRotation/SpiceRotation.cpp [hunk 2]
```diff
@@ -1222,15 +1303,32 @@ namespace Isis {
    * Set the rotation angles (phi, delta, and w) for the current time to define the
    * time-based matrix CJ. This method was created for unitTests and should not
    * be used otherwise.  It only works for cached data with a cache size of 1.
-   *
+   *  
+   * // FIXME - MAYBE THIS SHOULD BE IN THE UNITTEST THEN.... (see comment)
```
Comment (jessemapel) [inline isis/src/base/objs/SpiceRotation/SpiceRotation.cpp:1307]: Why is this here?

Comment (krlberry) [reply isis/src/base/objs/SpiceRotation/SpiceRotation.cpp:1307]: Thanks for catching this. This is a note I made to myself that I didn't end up changing. There's a comment in this function that it was only added for unit testing and shouldn't be used for anything else... so I thought it might make sense to just subclass SpiceRotation, add this function, and use that for the unit test. Then I realized that this is going to be gtest converted at some point anyway, and going to the extra effort of moving this probably wasn't worth it.

Comment (krlberry) [reply isis/src/base/objs/SpiceRotation/SpiceRotation.cpp:1307]: So, I'm just going to delete the comment.

**Comment by jessemapel** [comment]
Closed for now. Will re-open once we get more work on this.
